tf2_eigen/CMakeLists.txt

```diff
@@ -1,29 +1,39 @@
 cmake_minimum_required(VERSION 3.5)
 project(tf2_eigen)
 
-find_package(catkin REQUIRED COMPONENTS
-  cmake_modules
-  geometry_msgs
-  tf2
-)
-
-find_package(Eigen REQUIRED)
-include_directories(${EIGEN_INCLUDE_DIRS})
-add_definitions(${EIGEN_DEFINITIONS})
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wextra")
+endif()
 
-include_directories(include)
+find_package(ament_cmake REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_ros REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS include ${catkin_INCLUDE_DIRS}
-  DEPENDS eigen)
+find_package(Eigen3 QUIET)
+if(NOT Eigen3_FOUND)
+  message("Did not find Eigen3 on system. Looking manually.")
+  include(${tf2_eigen_SOURCE_DIR}/config/FindEigen3.cmake)
+endif()
+set(Eigen_INCLUDE_DIR "${EIGEN3_INCLUDE_DIR}")
+message("Using Eigen3 include dirs: ${Eigen_INCLUDE_DIR}")
 
 install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
-
-
-if(CATKIN_ENABLE_TESTING)
-
-  catkin_add_gtest(tf2_eigen-test test/tf2_eigen-test.cpp)
-  target_link_libraries(tf2_eigen-test ${catkin_LIBRARIES} ${GTEST_LIBRARIES})
-
+        DESTINATION include/${PROJECT_NAME})
+
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
+  ament_add_gtest(tf2_eigen-test test/tf2_eigen-test.cpp)
+  if(TARGET tf2_eigen-test)
+    target_include_directories(tf2_eigen-test PUBLIC
+      include
+      ${tf2_INCLUDE_DIRS}
+      ${tf2_ros_INCLUDE_DIRS}
+      ${Eigen_INCLUDE_DIR}
+    )
+  endif()
 endif()
+
+ament_export_include_directories(include)
+ament_export_dependencies(tf2_ros)
```

Comment:
Because tf2_eigen now depends on tf2_ros, we have to export the dependencies so downstream users get the right include headers.  This is done here.  This is new.